### PR TITLE
Feature: FilterBar in SwiftUI

### PR DESCRIPTION
**Pull request by renanmassaroto**
## Description

Adds FilterBar in SwiftUI

Fixes # (issue)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Screenshots (if appropriate):

https://github.com/ocean-ds/ocean-ios/assets/114941235/0d70352f-0582-470e-9634-ac01e063a22c

## Checklist:

- [X] I have read the [CONTRIBUTING](CONTRIBUTING.md) document
- [X] I have performed a self-review of my own code
- [X] My changes generate no new warnings
- [X] I have made corresponding changes to the documentation (if appropriate)
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] All new and existing tests passed

### CODE REVIEW

File: OceanComponents/Classes/Components SwiftUI/FilterBar/OceanSwiftUI+FilterBar.swift [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+                                ForEach(group.options, id: \.title) { option in
+                                    itemView(option: option, group: group)
+                                }
+                            }
+
+                            if parameters.groups.last?.id != group.id {
+                                Divider { divider in
+                                    divider.parameters.axis = .vertical
+                                }
+                                .frame(height: Constants.itemHeight)
```
Comment (vromeiro) [inline OceanComponents/Classes/Components SwiftUI/FilterBar/OceanSwiftUI+FilterBar.swift:91]: acho q pode usar o parametro q já tem no divider:
divider.parameters.size = Constants.itemHeight

File: OceanComponents/Classes/Components SwiftUI/FilterBar/OceanSwiftUI+FilterBar.swift [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+        private func itemView(option: FilterBarOption, group: FilterBarGroup) -> some View {
+            HStack {
+                Spacer(minLength: Ocean.size.spacingStackXs)
+
+                if let icon = option.leadingIcon {
+                    imageView(icon: icon, option: option)
+                    Spacer(minLength: Ocean.size.spacingStackXxxs)
+                }
+
+                Text(option.label)
```
Comment (vromeiro) [inline OceanComponents/Classes/Components SwiftUI/FilterBar/OceanSwiftUI+FilterBar.swift:111]: nao da pra usar o nosso Typography?

File: OceanComponents/Classes/Components SwiftUI/FilterBar/OceanSwiftUI+FilterBarBuilder.swift [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+//  OceanSwiftUI+FilterBarBuilder.swift
+//  OceanComponents
+//
+//  Created by Renan Massaroto on 09/01/24.
+//
+
+import OceanTokens
+
+extension OceanSwiftUI.FilterBar {
+    public static func filterBar(builder: OceanSwiftUI.FilterBar.Builder? = nil) -> OceanSwiftUI.FilterBar {
```
Comment (vromeiro) [inline OceanComponents/Classes/Components SwiftUI/FilterBar/OceanSwiftUI+FilterBarBuilder.swift:11]: precisa desse builder?
acho interessante só se tivermos vários tipos de filterBar pq colocar OceanSwiftUI.FilterBar.filterBar fica estranho...

**Review by vromeiro — APPROVED** [review]

File: OceanComponents/Classes/Components SwiftUI/FilterBar/OceanSwiftUI+FilterBar.swift (hunk 1 above)
Comment (renanmassaroto) [reply OceanComponents/Classes/Components SwiftUI/FilterBar/OceanSwiftUI+FilterBar.swift]: Se eu usar o divider.parameters.size a largura do Divider fica enorme, mas descobri que na verdade eu não preciso. Só preciso de uma padding pra dar margem do Divider para a Stack. Então removi o size do frame e coloquei o padding.

File: OceanComponents/Classes/Components SwiftUI/FilterBar/OceanSwiftUI+FilterBar.swift (hunk 2 above)
Comment (renanmassaroto) [reply OceanComponents/Classes/Components SwiftUI/FilterBar/OceanSwiftUI+FilterBar.swift]: Opa, passou batido! Feito!

File: OceanComponents/Classes/Components SwiftUI/FilterBar/OceanSwiftUI+FilterBarBuilder.swift (hunk 3 above)
Comment (renanmassaroto) [reply OceanComponents/Classes/Components SwiftUI/FilterBar/OceanSwiftUI+FilterBarBuilder.swift:11]: Removido!

**Pull request merged**